feat: Added a footer [#2]

yyjlincoln opened the pull request:
# Feature: Added a footer

## Intuition
A footer plays an important part in web design. It can serve as a sitemap that provides easy navigation between pages.

## Implementation
- We added new files, including `footer.html` and `footer.scss` to construct the UI.
- We added references to `footer.html` in `index` and `default` so that every page has the footer.

## Implementation Notes
- A body tag is automatically generated when the page is rendered (although it's unclear whether it was by the browser or by Jekyll); hence body tags in `default.html` and `index.html` are replaced with `<div class="main-content">`
- To provide a better visual, we introduced a minimum height of `100vh` for elements with class `.main-content`.

## Authors
- @yyjlincoln and @Dhruv-Sachdev1313

## PR Checklist
- [x] My Pod Leader knows I'm working on this Pull Request
- [x] I've explained what the Pull Request is adding.
- [x] I've explained why this is important.

## Additional Info
This pull request closes #2.

ditsuke commented on _layouts/default.html:
```diff
@@ -2,10 +2,15 @@
-<body>
-    {{ content }}
-</body>
+<div class="main-content">
```
How about `<main>`
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/main

yyjlincoln replied:
That sounds better. I'll do it now.

yyjlincoln replied:
Done! Feel free to approve the PR if you think it's ready to go.

ditsuke replied:
Awesome!

ditsuke reviewed (commented):
LGTM! Left a comment about using `<main>`, more semantic but nothing serious.

Aniket762 reviewed (commented):
Looks great!

Dhruv-Sachdev1313 reviewed (approved):
Looks good to me!

gmelodie commented on _layouts/page.html:
```diff
@@ -6,7 +6,7 @@
         <div class="logo">
             <a href="/">
-                <img src="/assets/img/logo.svg"/>
+                <img src="/assets/img/logo.svg" />
```
:scissors:

gmelodie commented on _layouts/default.html:
```diff
@@ -2,10 +2,15 @@
-<body>
-    {{ content }}
-</body>
+<main>
+    <div>
+        {{ content }}
+    </div>
+</main>
```
As this is not the purpose of this PR, leave it as it was

gmelodie commented on _layouts/page.html:
```diff
@@ -17,4 +17,5 @@
     {{ content }}
-</div>
\ No newline at end of file
+</div>
+
```
:scissors:

gmelodie commented on _sass/all.scss:
```diff
@@ -55,6 +56,10 @@ a {
     color: black;
 }
 
+main {
+  min-height: 100vh;
+}
+
```
Change `main` to `body`

yyjlincoln replied:
We tried. I believe Jekyll automatically generates the body tag that includes the footer element, so it did not achieve what we wanted. We'll give it another try.

gmelodie commented on index.html:
```diff
@@ -2,16 +2,24 @@
 ---
 <!DOCTYPE html>
 <html lang="en">
+
```
:scissors:

gmelodie commented on index.html:
```diff
@@ -2,16 +2,24 @@
 
-<body>
-    {% include navbar.html %}
+<main>
+
+    <div>
+        {% include navbar.html %}
+
+        {% include profile.html %}
+
+        {% include description.html %}
+
+        {% include projects.html %}
+
+    </div>
+
+</main>
```
Again, leave it as it was before

gmelodie reviewed (changes requested):
A PR should only fix the issue at hand, no more, no less. Changing more than what's required might cause unexpected bugs and merge conflicts and plus these other changes were not discussed with the rest of the team.

yyjlincoln commented:
Fixed up the history commit messages just now & formatting to minimize changes to main.
Merging to `main`:
- all checks have passed
- at least two approvals
- fixed up the suggested changes
